src/rust/bitbox02-rust/src/hww/api/cardano/sign_transaction/cbor.rs

```diff
@@ -165,6 +165,45 @@ pub fn encode_transaction_body<W: Write>(
                     encode_stake_credential(&mut encoder, keypath)?;
                     encoder.bytes(pool_keyhash)?;
                 }
+                certificate::Cert::VoteDelegation(certificate::VoteDelegation {
+                    keypath,
+                    r#type,
+                    drep_credhash,
+                }) => {
+                    encoder.array(3)?.u8(9)?;
+                    encode_stake_credential(&mut encoder, keypath)?;
+                    let drep_type = certificate::vote_delegation::CardanoDRepType::try_from(*r#type)?;
+                    match drep_type {
+                        certificate::vote_delegation::CardanoDRepType::KeyHash => {
+                            encoder.array(2)?.u8(0)?;
```

Comment:
I have no good of way of verifying this is correct, so we'll see in an end-to-end test, but

https://github.com/IntersectMBO/cardano-ledger/blob/cardano-ledger-conway-1.12.0.0/eras/conway/impl/cddl-files/conway.cddl#L273

> vote_deleg_cert = (9, stake_credential, drep)

I guess the `encoder.array(3?).u8(9)?` above starts the array and adds the first element. Shouldn't the `stake_credential` and `drep` now be added directly as two elements? Here it seems like you are adding them as an array of 2 elements.

Have you verified somehow that the cbor output here is correct? Could very well be that I am just mis-reading this, it's a bit hairy :sweat_smile:

Reply:
Nevermind, I just realized the stake_credential part is itself an array of 2!